pkg/redirectpolicy/redirectpolicy.go

```diff
@@ -296,3 +297,44 @@ func (config *LRPConfig) checkNamespace(namespace string) bool {
 	}
 	return true
 }
+
+func (config *LRPConfig) GetModel() *models.LRPSpec {
+	if config == nil {
+		return nil
+	}
+
+	var feType, lrpType string
+	switch config.frontendType {
+	case frontendTypeUnknown:
+		feType = "unknown"
+	case svcFrontendAll:
+		feType = "clusterIP + all svc ports"
+	case svcFrontendNamedPorts:
+		feType = "clusterIP + named ports"
+	case svcFrontendSinglePort:
+		feType = "clusterIP + port"
+	case addrFrontendSinglePort:
+		feType = "IP + port"
+	case addrFrontendNamedPorts:
+		feType = "IP + named ports"
+	}
+
+	switch config.lrpType {
+	case lrpConfigTypeNone:
+		lrpType = "none"
+	case lrpConfigTypeAddr:
+		lrpType = "addr"
+	case lrpConfigTypeSvc:
+		lrpType = "svc"
+	}
+
+	spec := &models.LRPSpec{
+		UID: string(config.uid),
+		Name: config.id.Name,
+		Namespace: config.id.Namespace,
+		FrontendType: feType,
+		LrpType: lrpType,
+		EndpointSelector: config.backendSelector.CachedLabelSelectorString,
+	}
```

Comment:
Please run `gofmt` on this change as per the GitHub action failure.

Reply:
Done.